visibility.py

```diff
@@ -0,0 +1,287 @@
+  airmassformat = []
+  for t in range(airmass2.size): airmassformat.append("%2.2f" % airmass2[t])
+  ax22.set_yticklabels(airmassformat, rotation=90)
+  ax22.tick_params(axis="y", pad=10, labelsize=10)
+  plt.text(1.045,-0.04, "Spherical+Alt", transform=ax.transAxes, ha='left', va='top', \
+           fontsize=10, rotation=90)
+
+  ax3 = ax.twiny()
+  ax3.set_frame_on(True)
+  ax3.patch.set_visible(False)
+  ax3.xaxis.set_ticks_position('bottom')
+  ax3.xaxis.set_label_position('bottom')
+  ax3.spines['bottom'].set_position(('outward', 50))
+  ax3.spines['bottom'].set_color('k')
+  ax3.spines['bottom'].set_visible(True)
+
+  ltime, ldiff = pyasl.localtime.localTime(utchours, np.repeat(obs['longitude'], len(utchours)))
+  jdltime = jdhours - ldiff/24.
+  ax3.set_xticks(jdltime)
+  ax3.set_xticklabels(utchours)
+  ax3.set_xlim([axrange[0],axrange[1]])
+  ax3.set_xlabel("Local time [hours]")
+
+  ax.set_ylim([0, 91])
+  ax.yaxis.set_major_locator(MultipleLocator(15))
+  ax.yaxis.set_minor_locator(MultipleLocator(5))
+  yticks = ax.get_yticks()
+  ytickformat = []
+  for t in range(yticks.size): ytickformat.append(str(int(yticks[t]))+r"$^\circ$")
+  ax.set_yticklabels(ytickformat, fontsize=16)
+  ax.set_ylabel("Altitude", fontsize=18)
+  yticksminor = ax.get_yticks(minor=True)
+  ymind = np.where( yticksminor % 15. != 0. )[0]
+  yticksminor = yticksminor[ymind]
+  ax.set_yticks(yticksminor, minor=True)
+  m_ytickformat = []
+  for t in range(yticksminor.size): m_ytickformat.append(str(int(yticksminor[t]))+r"$^\circ$")
+  ax.set_yticklabels(m_ytickformat, minor=True)
+  ax.set_ylim([0, 91])
+  
+  ax.yaxis.grid(color='gray', linestyle='dashed')
+  ax.yaxis.grid(color='gray', which="minor", linestyle='dotted')
+  ax2.xaxis.grid(color='gray', linestyle='dotted')
+
+  plt.text(0.5,0.95,"Visibility on %s" % date.date(), \
+           transform=fig.transFigure, ha='center', va='bottom', fontsize=20)
+
+  if plotLegend:
+    line1 = matplotlib.lines.Line2D((0,0),(1,1), color='#FDB813', linestyle="-", linewidth=2)
+    line2 = matplotlib.lines.Line2D((0,0),(1,1), color='#BEBEBE', linestyle="-", linewidth=2)
+    line3 = matplotlib.lines.Line2D((0,0),(1,1), color='k', linestyle="-", linewidth=2)
+
+    lgd2 = plt.legend((line1,line2,line3),("day","twilight","night",), \
+                        bbox_to_anchor=(0.88, 0.13), loc='best', borderaxespad=0.,prop={'size':12}, fancybox=True)
+    lgd2.get_frame().set_alpha(.5)
+
+  obsco = "Obs coord.: %8.4f$^\circ$, %8.4f$^\circ$, %4d m" % \
+          (obs['longitude'], obs['latitude'], obs['altitude'])
+
+  plt.text(0.01,0.97, obsco, transform=fig.transFigure, ha='left', va='center', fontsize=10)
+  plt.text(0.01,0.95, obs['name'], transform=fig.transFigure, ha='left', va='center', fontsize=10)
+  
+  if print2file:
+    pass
+    # plt.savefig(outfile, format="png", dpi=300)
+  else:
+    plt.show()
+
+
+
+if __name__ == '__main__':
+  import datetime as dt
+
+  ids = ['HD41248', 'HD219828']
+  targets = []
```

Comment:
Can be written as `targets = [{'name': i, 'coord': SkyCoord.from_name(i)} for i in ids]` :) But not important.